tests/client/test_redirects.py

```diff
@@ -155,9 +156,43 @@ async def body():
         return b"HTTP/1.1", 200, b"OK", [], ByteStream(b"Hello, world!")
 
 
+class SyncMockTransport(httpcore.SyncHTTPTransport):
```

Comment:
Same here.

Reply:
I would suggest the same as above, this one is a bit longer so it adds a bit more noise, but again no big deal.

Reply:
For this one it's a bit harder, since there are a couple of `part async for part in stream` inside. I tried changing those to `part for part in stream` but it fails with `RuntimeError: Attempted to call a sync iterator on an async stream.`.

It's only really used in `redirect_body_target` (in the other usages the content is discarded). Is there any way to fetch the stream independently from the context (async/sync) we're in?

Reply:
Nevermind, I used the example in [custom transports](https://www.python-httpx.org/advanced/#custom-transports) and it's working now :)